opte/src/engine/port.rs

```diff
@@ -561,12 +576,17 @@ impl Port {
     ///
     /// This command is valid for any [`PortState`].
     pub fn dump_layer(&self, name: &str) -> Result<ioctl::DumpLayerResp> {
+        let state = self.state.lock();
+
         for l in &*self.layers {
             if l.name() == name {
                 return Ok(l.dump());
             }
         }
 
+        // Perhaps this is silly, but I want to make sure the state
+        // lock is held for the execution of this method.
+        drop(state);
```

Comment:
I think this is a solid way to do that. The other way I've seen is to introduce an inner scope:

```rust
    {
        let state = self.state.lock();
        // Do stuff with state
    }
    // Lock is dropped here.
    Err(OpteError::LayerNotFound(name.to_string()))
```

Reply:
I don't think that does what I'm after. The point is I'm not doing anything with `state`, but what the lock to be help. I wasn't sure if the compile could detect this and just optimize it away or something (also I want to silence the warning). I'll just leave it as-is for now.

Reply:
Yeah, that's fair, I'm not sure what happens to a lock guard that you never access. Maybe an argument for locking the data itself, to avoid worrying about that. FWIW, you can silence the "unused variable" warnings by prefixing the name with `_`, e.g., `let _state = self.state.lock();`.

Leaving it is fine, I agree it works.